src/erasure-code/ErasureCode.cc

```diff
@@ -215,15 +215,25 @@ int ErasureCode::encode(const set<int> &want_to_encode,
 {
   unsigned int k = get_data_chunk_count();
   unsigned int m = get_chunk_count() - k;
-  bufferlist out;
+  if (!encoded || !encoded->empty()){
+      return -EINVAL;
+    }
   int err = encode_prepare(in, *encoded);
   if (err)
     return err;
-  encode_chunks(want_to_encode, encoded);
+std::map<int, bufferptr> in_shards;
+    std::map<int, bufferptr> out_shards;
+    for (auto&& [shard, list] : *encoded) {
+      auto bp = list.begin().get_current_ptr();
```

Comment:
Might be dangerous. What is the input `list` has more than 1 buffer?
At least an assertion on `list.get_num_buffers() == 1` would be needed.
Also, instead of constructing the iterator, you could simply call `list.front()`.

Ideally these `map-of-list`-taking variant would be replaced with `map-of-bp`.